lib/ansible/modules/network/cloudengine/ce_lacp.py

```diff
@@ -0,0 +1,548 @@
+        self.param = dict()
+
+        self.state = self.module.params['state']
+
+        # state
+        self.changed = False
+        self.updates_cmd = list()
+        self.results = dict()
+        self.proposed = dict()
+        self.existing = dict()
+        self.end_state = dict()
+
+    def __init_module__(self):
+        """ init module """
+
+        self.module = AnsibleModule(
+            argument_spec=self.spec,
+            mutually_exclusive=[['trunk_id', 'global_priority']],
+            required_one_of=[['trunk_id', 'global_priority']],
+            required_together=[['member_if', 'priority']],
+            supports_check_mode=True)
+
+    def check_params(self):
+        """
+        :return:
+        """
+        for key in self.module.params.keys():
+            if key in LACP.keys() and self.module.params[key] is not None:
+                self.param[key] = self.module.params[key]
+        msg = check_param(self.param)
+        # if self.param.get('fast_timeout') is not None and self.param.get('timeout_type') is None:
+        # self.param['timeout_type'] = 'Fast'
+        if msg != 'ok':
+            self.module.fail_json(msg=msg)
+
+    def get_existing(self):
+        """
+        :return:
+        """
+        xml_str = bulid_xml(self.param)
+        xml = get_nc_config(self.module, xml_str)
+        return xml_to_dict(xml)
+
+    def get_proposed(self):
+        """
+        :return:
+        """
+        proposed = dict(state=self.state)
+        proposed.update(self.param)
+        return proposed
+
+    def get_end_state(self):
+        """
+        :return:
+        """
+        xml_str = bulid_xml(self.param)
+        xml = get_nc_config(self.module, xml_str)
+        return xml_to_dict(xml)
+
+    def work(self):
+        """worker"""
+
+        self.check_params()
+        existing = self.get_existing()
+        proposed = self.get_proposed()
+
+        # deal present or absent
+        if self.state == "present":
+            operation = 'merge'
+        else:
+            operation = 'delete'
+
+        xml_str = bulid_xml(self.param, operation=operation)
+        set_nc_config(self.module, xml_str)
+        end_state = self.get_end_state()
+
+        self.results['proposed'] = proposed
+        self.results['existing'] = existing
+        self.results['end_state'] = end_state
+        updates_cmd = compare_config(self.module, existing, end_state)
+        self.results['updates'] = updates_cmd
+        if updates_cmd:
+            self.results['changed'] = True
+        else:
+            self.results['changed'] = False
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """Module main"""
+
+    argument_spec = dict(
+        mode=dict(required=False,
+                  choices=['Manual', 'Dynamic', 'Static'],
+                  type='str'),
+        trunk_id=dict(required=False, type='str'),
+        preempt_enable=dict(required=False, choices=['true', 'false']),
```

Comment:
should be rewritten
```suggestion
        preempt_enable=dict(type='bool'),
```
etc for all params where needed